CRM/Event/Form/Registration/Confirm.php

```diff
@@ -252,6 +252,7 @@ public function buildQuickForm() {
         $this->assign('pay_later_receipt', '');
         // @fixme These functions all seem to do similar things but take one away and the house of cards falls down..
         $this->assignPaymentProcessor($this->_values['event']['is_pay_later']);
+        $this->preProcessPaymentOptions();
```

Comment:
This is required only after the form is submitted to repopulate form fields so that eg. credit card fields can be retrieved via getSubmittedValue() from the ThankYou page. Otherwise they are lost.

Reply:
@mattwire maybe we need an incode comment to that effect?

Reply:
@eileenmcnaughton good idea. Added